packages/pulumi-aws/src/components/CloudFrontBucket.ts

```diff
@@ -0,0 +1,70 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import { PulumiInputValue } from "../types";
+import { createResource, ResourceOverride } from "./createResource";
+
+type OriginConfig = PulumiInputValue<
+    PulumiInputValue<aws.cloudfront.DistributionArgs["origins"]>[number]
+>;
+
+export interface CloudFrontBucketConfig {
+    bucketConfig: ResourceOverride<typeof aws.s3.Bucket>;
+}
+
+export class CloudFrontBucket extends pulumi.ComponentResource {
+    public readonly s3Bucket: aws.s3.Bucket;
+    public readonly origin: OriginConfig;
+    public readonly originIdentity: aws.cloudfront.OriginAccessIdentity;
+
+    constructor(name: string, config?: CloudFrontBucketConfig) {
+        super("webiny:aws:CloudFrontBucket", name);
+
+        // Origin Identity is a kind of AWS user that represents Cloudfront distribution
+        // We can add IAM policies to it later, to allow accessing private S3 bucket
+        this.originIdentity = new aws.cloudfront.OriginAccessIdentity(`${name}-origin-identity`);
+
+        this.s3Bucket = createResource(aws.s3.Bucket, name, config?.bucketConfig, {
+            acl: aws.s3.CannedAcl.Private,
+            forceDestroy: true
+        });
```

Comment:
This function you use instead of calling `new aws.s3.Bucket`. This is a syntactic sugar really. It will run config override logic and call constructor inside. Provides some nice typing.